server/resources/products/productsController.js

```diff
@@ -42,16 +42,62 @@ exports.list = (req, res) => {
   }
 }
 
+exports.listByValues = (req, res) => {
+  /**
+   * returns list of products queried from the array of _id's passed in the query param
+   * 
+   * NOTES:
+   * 1) looks like the best syntax for this is, "?id=1234&id=4567&id=91011"
+   *    still a GET, and more or less conforms to REST uri's
+   *    additionally, node will automatically parse this into a single array via "req.query.id"
+   * 2) node default max request headers + uri size is 80kb. 
+   *    experimentation needed to determie what the max length of a list we can do this way is
+   * TODO: server side pagination
+   */ 
+
+  if(!req.query[req.params.refKey]) {
+    // make sure the correct query params are included
+    res.send({success: false, message: `Missing query param(s) specified by the ref: ${req.params.refKey}`});
+  } else {
+    // as in listByRef below, attempt to query for matching ObjectId keys first. ie, if "user" is passed, look for key "_user" before key "user"
+    Product.find({["_" + req.params.refKey]: {$in: [].concat(req.query[req.params.refKey]) }}, (err, products) => {
+        if(err || !products) {
+          res.send({success: false, message: `Error querying for products by ${["_" + req.params.refKey]} list`, err});
+        } else if(products.length == 0) {
+          Product.find({[req.params.refKey]: {$in: [].concat(req.query[req.params.refKey]) }}, (err, products) => {
+            if(err || !products) {
+              res.send({success: false, message: `Error querying for products by ${[req.params.refKey]} list`, err});
+            } else {
+              res.send({success: true, products});
+            }
+          })
+        } else  {
+          res.send({success: true, products});
+        }
+    })
+  }
+}
+
 exports.listByRef = (req, res) => {
   /**
-   * NOTE: This let's us querey ANY pointer by passing in a refKey and refId
+   * NOTE: This let's us query by ANY string or pointer key by passing in a refKey and refId
+   * TODO: server side pagination
    */
-  let query = {
-    ["_" + req.params.refKey]: req.params.refId
-  }
-  Product.find(query, (err, products) => {
+  Product.find({["_" + req.params.refKey]: [req.params.refId]}, (err, products) => {
```

Comment:
why don't we just pass in the refKey with the `_` in the call?  api call would be `https://yote.com/api/posts/by-_author/5xyz`